axiom/nr_php_packages.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef NR_PHP_PACKAGES_HDR
+#define NR_PHP_PACKAGES_HDR
+
+#include "nr_log_event.h"
+#include "util_random.h"
+#include "util_vector.h"
+#include "util_hashmap.h"
+
+typedef struct _nr_php_package_t {
+  char* package_name;
+  char* package_version;
+} nr_php_package_t;
+
+typedef struct _nr_php_packages_t {
+  nr_hashmap_t* data;
+} nr_php_packages_t;
+
+/*
+ * Purpose : Create a new php package
+ *
+ * Params  : 1. Package name
+ *           2. Package version
+ *
+ * Returns : A php package that has a name and version. If
+ *           nr_php_packages_add_package() is not called, then it must be freed
+ *           by nr_php_package_destroy()
+ */
+extern nr_php_package_t* nr_php_package_create(char* name, char* version);
+
+/*
+ * Purpose : Destroy/free php package
+ *
+ * Params  : The php package to free
+ *
+ * Returns : Nothing, it is void
+ */
+extern void nr_php_package_destroy(nr_php_package_t* p);
+
+/*
+ * Purpose : Allocate memory for new collection that will hold packages
+ *
+ * Returns : A collection that is allocated of type nr_php_packages_t
+ */
+extern nr_php_packages_t* nr_php_packages_create(void);
+
+/*
+ * Purpose : Add new php package to collection. If a package with the same key
+ *           but different value is added, then the old value will be freed and
+ *           the newer value will be kept. If a duplicate package with the same
+ *           key and value is added, no action will be taken. This function also
+ *           frees the package being added, so the user does not need to call
+ *           nr_php_package_destroy()
```

Comment:
I would rewrite this to indicate the caller is not responsible for destroying the package whether or not there is a name collision.

Reply:
updated: [90ece38](https://github.com/newrelic/newrelic-php-agent/pull/754/commits/90ece38106c8b0fa9deb82bde4606dd92cd413bf)